zeppelin-zengine/src/test/java/org/apache/zeppelin/notebook/NotebookTest.java

```diff
@@ -150,13 +150,14 @@ public void testSchedule() throws InterruptedException, IOException{
 		note.setConfig(config);
 		notebook.refreshCron(note.id());
 		Thread.sleep(1*1000);
-		dateFinished = p.getDateFinished();
-		assertNotNull(dateFinished);
-
+		
 		// remove cron scheduler.
 		config.put("cron", null);
 		note.setConfig(config);
 		notebook.refreshCron(note.id());
+		Thread.sleep(1000);
+		dateFinished = p.getDateFinished();
+		assertNotNull(dateFinished);
 		Thread.sleep(1*1000);
 		assertEquals(dateFinished, p.getDateFinished());
```

Comment:
To me it looks like that this change only postpones the already existing check 1 second to make sure that `notebook.refreshCron(note.id());` was executed in any case. It should cause the test to pass in any case, even if the first second was not enough time for cron to shut down.